src/components/ChangesNotAppliedModal.jsx

```diff
@@ -1,37 +1,51 @@
 import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 import PropTypes from 'prop-types';
 import {
   Modal, Typography, Space, Button,
 } from 'antd';
+import { useRouter } from 'next/router';
+import { discardChangedQCFilters, navigateFromProcessingTo } from '../redux/actions/experimentSettings';
+import { runPipeline } from '../redux/actions/pipeline';
 
 import { getUserFriendlyQCStepName } from '../utils/qcSteps';
 
 const { Text } = Typography;
 
 const ChangesNotAppliedModal = (props) => {
-  const {
-    steps, visible, onRun, onDiscard, onCancel,
-  } = props;
+  const { experimentId } = props;
+
+  const router = useRouter();
+  const dispatch = useDispatch();
+
+  const { changedQCFilters, navigationPath } = useSelector((state) => state.experimentSettings.processing.meta);
 
   return (
     <Modal
-      visible={visible}
+      visible={navigationPath}
       title='Changes not applied'
-      onCancel={onCancel}
+      onCancel={() => dispatch(navigateFromProcessingTo(''))}
       footer={(
         <Space size='large' style={{ display: 'flex', justifyContent: 'center' }}>
           <Button
             type='primary'
             key='run'
-            onClick={onRun}
+            onClick={() => {
+              dispatch(runPipeline(experimentId));
+              dispatch(navigateFromProcessingTo(''));
+            }}
             style={{ width: '100px' }}
           >
             Run
           </Button>
           <Button
             type='primary'
             key='discard'
-            onClick={onDiscard}
+            onClick={() => {
```

Comment:
Same with this, was was it moved into the modal?

Reply:
Same as above